jupyterlab_wipp/src/handler.ts

```diff
@@ -0,0 +1,46 @@
+import { URLExt } from '@jupyterlab/coreutils';
+
+import { ServerConnection } from '@jupyterlab/services';
+
+/**
+ * Call the API extension
+ *
+ * @param endPoint API REST end point for the extension
+ * @param init Initial values for the request
+ * @returns The response body interpreted as JSON
+ */
+export async function requestAPI<T>(
+  endPoint = '',
+  init: RequestInit = {}
+): Promise<T> {
+  // Make request to Jupyter API
+  const settings = ServerConnection.makeSettings();
+  const requestUrl = URLExt.join(
+    settings.baseUrl,
+    'wipp', // API Namespace
+    endPoint
+  );
+
+  let response: Response;
+  try {
+    response = await ServerConnection.makeRequest(requestUrl, init, settings);
+  } catch (error) {
+    throw new ServerConnection.NetworkError(error);
+  }
+
+  let data: any = await response.text();
+
+  if (data.length > 0) {
+    try {
+      data = JSON.parse(data);
+    } catch (error) {
+      console.log('Not a JSON response body.', response);
```

Comment:
Can we throw an error instead of writing directly to console?

Reply:
We can, but JupyterLab and existing extensions use console heavily for all sorts of logging and error reporting. for example, see screenshot of console when loading notebooks in CI: 
![image](https://user-images.githubusercontent.com/8834829/124184386-58712580-da6e-11eb-9072-c1066ffe2c67.png)

This particular line (and the whole file) is coming from an extension cookiecutter: 
https://github.com/jupyterlab/extension-cookiecutter-ts/blob/057d135d10e62570abcd9b0f1db92022b5ab6faf/%7B%7Bcookiecutter.python_name%7D%7D/src/handler.ts#L37

We can either open PR to change the line there, or only change it in this extension, or don't change it all.

Reply:
Also, I compared the effect of throwing new error vs. calling console.error:
```ts
console.error(`The jupyterlab_wipp server extension appears to be missing.\n${reason}`);
throw new Error(`The jupyterlab_wipp server extension appears to be missing.\n${reason}`);
```
They look very similar in the console, but the latter causes unhandled exception since there is nothing up the stack to catch it.
![image](https://user-images.githubusercontent.com/8834829/124190643-78591700-da77-11eb-97e6-9c36d39a70bc.png)
There is no difference to the end user unless they open a console.